lib/engine/game/g_18_chesapeake.rb

```diff
@@ -34,11 +34,20 @@ def init_share_pool
         Engine::G18Chesapeake::SharePool.new(self)
       end
 
+      def preprocess_action(action)
+        case action
+        when Action::LayTile
+          queue_log!('special_tile_lay') do
+            check_special_tile_lay(action, baltimore)
+            check_special_tile_lay(action, columbia)
+          end
+        end
+      end
+
       def action_processed(action)
         case action
         when Action::LayTile
-          check_special_tile_lay(action, columbia)
-          check_special_tile_lay(action, baltimore)
+          flush_log!('special_tile_lay')
```

Comment:
With the timing changes, having this check after the action processed meant the action was already used up and removed by the time `action_processed` is called, but it looked weird with the log lines about losing the ability to lay happening before the tile lay.

This initially was behaving strangely with the first place `queue_log!` was used (the phase change in `train.rb`), so I changed queued_log to be a hash that could accept different string descriptions and use them as keys for the log messages to display later.

Reply:
idk what happened but I found I was actually able to eliminate the string arg to `queue_log!`